code/modules/spells/spell_types/shapeshift.dm

```diff
@@ -27,7 +27,10 @@
 			for(var/path in possible_shapes)
 				var/mob/living/simple_animal/A = path
 				animal_list[initial(A.name)] = path
-			shapeshift_type = input(M, "Choose Your Animal Form!", "It's Morphing Time!", null) as anything in animal_list
+			var/new_shapeshift_type = input(M, "Choose Your Animal Form!", "It's Morphing Time!", null) as null|anything in animal_list
+			if(shapeshift_type)
+				return
```

Comment:
What gives? It looks like the prompt still fires, but it may ignore player input without any feedback. I think you should move the conditional to before the input call.